engine/engine_test.go

```diff
@@ -415,3 +418,77 @@ func TestGetDefaultConfigurations(t *testing.T) {
 		})
 	}
 }
+
+func TestSetupExchanges(t *testing.T) {
+	t.Parallel()
+	e := &Engine{
+		Config: &config.Config{Exchanges: []config.Exchange{{Name: testExchange}}},
+	}
+	assert.ErrorIs(t, e.SetupExchanges(), ErrNoExchangesLoaded)
+
+	settings := Settings{
+		CoreSettings: CoreSettings{
+			EnableAllPairs:     true,
+			EnableAllExchanges: true,
+		},
+		ExchangeTuningSettings: ExchangeTuningSettings{
+			EnableExchangeVerbose:          true,
+			EnableExchangeWebsocketSupport: true,
+			EnableExchangeAutoPairUpdates:  true,
+			DisableExchangeAutoPairUpdates: true,
+			HTTPUserAgent:                  "test",
+			HTTPProxy:                      "test",
+			HTTPTimeout:                    1,
+			EnableExchangeHTTPDebugging:    true,
+		},
+	}
+
+	// Test that adjusting settings induces dry run mode correctly
+	e.Settings = settings
+	assert.ErrorIs(t, e.SetupExchanges(), ErrNoExchangesLoaded)
+	assert.False(t, e.Settings.EnableDryRun)
+	e.Settings.CheckParamInteraction = true
+	assert.ErrorIs(t, e.SetupExchanges(), ErrNoExchangesLoaded)
+	assert.True(t, e.Settings.EnableDryRun)
+
+	// Test that overridden exchange inputs are handled correctly
+	testCases := []struct {
+		name           string
+		exchangeString string
+		expectedError  string
+	}{
+		{"Invalid exchange pair", "bob|jill", "exchange bob|jill not found"},
+		{"Single invalid exchange", "bob", "exchange bob not found"},
+		{"Mixed valid and invalid exchanges", "bob,bitstamp", "exchange bob not found"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			e.Settings = Settings{CoreSettings: CoreSettings{Exchanges: tc.exchangeString}}
```

Comment:
You can't share e.Settings with parallel and `e.Settings = {}` below either 💥🗞️ Causes data race issues:
```
 go test -run=TestSetupExchanges -count=1 -race
# github.com/thrasher-corp/gocryptotrader/engine.test
ld: warning: '/private/var/folders/6s/z1b0pygn0xb9lqbkghk2_btr0000gn/T/go-link-3044902326/000023.o' has malformed LC_DYSYMTAB, expected 98 undefined symbols to start at index 1626, found 95 undefined symbols starting at index 1626
==================
WARNING: DATA RACE
Write at 0x00c0004842d0 by goroutine 24:
  github.com/thrasher-corp/gocryptotrader/engine.TestSetupExchanges.func1()
      /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/engine/engine_test.go:468 +0x80
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0x184
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x40

Previous write at 0x00c0004842d0 by goroutine 22:
  github.com/thrasher-corp/gocryptotrader/engine.TestSetupExchanges.func1()
      /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/engine/engine_test.go:468 +0x80
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0x184
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x40

Goroutine 24 (running) created at:
  testing.(*T).Run()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x5e0
  github.com/thrasher-corp/gocryptotrader/engine.TestSetupExchanges()
      /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/engine/engine_test.go:466 +0x524
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0x184
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x40

Goroutine 22 (finished) created at:
  testing.(*T).Run()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x5e0
  github.com/thrasher-corp/gocryptotrader/engine.TestSetupExchanges()
      /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/engine/engine_test.go:466 +0x524
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0x184
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x40
==================
--- FAIL: TestSetupExchanges (0.27s)
    --- FAIL: TestSetupExchanges/Single_invalid_exchange (0.00s)
        testing.go:1399: race detected during execution of test
    --- FAIL: TestSetupExchanges/Mixed_valid_and_invalid_exchanges (0.00s)
        testing.go:1399: race detected during execution of test
FAIL
exit status 1
FAIL	github.com/thrasher-corp/gocryptotrader/engine	0.846s
```